dev/deps/spark-deps-hadoop-3.2

```diff
@@ -22,6 +22,8 @@ automaton-1.11-8.jar
 avro-1.8.2.jar
 avro-ipc-1.8.2.jar
 avro-mapred-1.8.2-hadoop2.jar
+bcpkix-jdk15on-1.60.jar
+bcprov-jdk15on-1.60.jar
```

Comment:
FYI, we already have this in `NOTICE-binary`.
```
This product optionally depends on 'Bouncy Castle Crypto APIs' to generate
a temporary self-signed X.509 certificate when the JVM does not provide the
equivalent functionality.  It can be obtained at:

  * LICENSE:
    * license/LICENSE.bouncycastle.txt (MIT License)
  * HOMEPAGE:
    * http://www.bouncycastle.org/
```

Reply:
@srowen . Could you give me some advice?

Reply:
There are a few parts here. First, that NOTICE statement is from Hadoop's NOTICE, so I'd copy whatever it says now, to update.

Second, if it's a first-class dependency now, it needs to have a line in LICENSE-binary and a copy of the license in licenses-binary/. It's MIT-licensed so should be OK.

Finally, BC is a special case because it's subject to crypto export laws. We will have to update http://www.apache.org/licenses/exports/ to say that it's again a dependency in 3.0. I can go figure that out again as and when this is merged.

Reply:
Oops. It sounds too much for me. Please help me after I merge this.

Reply:
For the record, the process for the last part is https://www.apache.org/dev/crypto.html#sources
I can do it afterwards, it's not hard.

However, hm, I wonder if Hadoop needs a similar disclosure at http://www.apache.org/licenses/exports/ ? It's possible that somehow it isn't distributed directly by Hadoop, but would be surprised if it's a first-class dep and it makes binary releases.

Maybe, eh, CC @steveloughran in case he knows anything about this angle.

Reply:
ooh, it is in the binary isn't it
```
hadoop-3.2.1 find . -print | grep bcp
./share/hadoop/yarn/lib/bcprov-jdk15on-1.60.jar
./share/hadoop/yarn/lib/bcpkix-jdk15on-1.60.jar
```

let me chase this up

Reply:
Thank you, @srowen and @steveloughran .
I will use this chance to learn this legal process. :)